packages/admin-frontend/components/edit_book/EditBookForm.js

```diff
@@ -0,0 +1,177 @@
+// @flow
+import * as React from 'react';
+import Link from 'next/link';
+import { Form, Field } from 'react-final-form';
+import Button from '@material-ui/core/Button/Button';
+import Grid from '@material-ui/core/Grid';
+import TextField from '@material-ui/core/TextField';
+import Typography from '@material-ui/core/Typography';
+import Select from '@material-ui/core/Select';
+import FormHelperText from '@material-ui/core/FormHelperText';
+
+import { saveBook } from '../../lib/fetch';
+import type { BookDetails } from '../../types';
+
+const PUBLISHING_STATUS = ['PUBLISHED', 'FLAGGED', 'UNLISTED'];
+const PAGE_ORIENTATIONS = ['PORTRAIT', 'LANDSCAPE'];
+
+type Props = {
+  book: BookDetails
+};
+
+type State = {
+  book: BookDetails
+};
+
+export default class EditBookForm extends React.Component<Props, State> {
+  state = {
+    book: this.props.book
+  };
+
+  handleSubmit = (content: BookDetails) => {
+    this.updateBook(content);
+  };
+
+  updateBook = async (content: BookDetails) => {
+    const result = await saveBook(content);
+    if (result.isOk) {
+      this.setState({ book: content });
+    }
+  };
+
+  render() {
+    const book = this.state.book;
+
+    return (
+      <div>
+        {' '}
+        {book && (
+          <Typography variant="headline" component="h1" gutterBottom>
+            Editing book:{' '}
+            <Link href={`/${book.language.code}/books/details/${book.id}`}>
+              <a>{book.title}</a>
+            </Link>
+          </Typography>
+        )}
+        <Form
+          initialValues={book}
+          onSubmit={this.handleSubmit}
+          validate={handleValidate}
+          render={({ handleSubmit, pristine, form, invalid }) => (
+            <form>
+              <Grid container spacing={24} direction="column">
```

Comment:
Suggestion: Simplify spacing by removing the grid thingy and set margin on the TextFields instead? See https://material-ui.com/api/text-field/

Reply:
fixed